Test_buildCAFromSecret is flaky

`Test_buildCAFromSecret` failed at least twice recently:

```
✓  cmd/manager (37ms) (coverage: 5.6% of statements)
=== Failed
=== FAIL: pkg/controller/common/certificates Test_buildCAFromSecret/valid_secret (0.00s)
    ca_reconcile_test.go:416: CaFromSecrets() got = &{0xc000265030 0xc0007f6008}, want &{0xc000264ee0 0xc00025b808}
=== FAIL: pkg/controller/common/certificates Test_buildCAFromSecret (0.66s)
DONE 3615 tests, 2 failures in 234.739s
make: *** [Makefile:150: unit-xml] Error 1
🚨 Error: The command exited with status 2
user command error: exit status 2
```

--- a/pkg/controller/common/certificates/ca_reconcile_test.go
+++ b/pkg/controller/common/certificates/ca_reconcile_test.go
@@ -149,7 +149,7 @@ func Test_canReuseCA(t *testing.T) {
 func privateKeysEqual(t *testing.T, actual, expected crypto.Signer) {
 	t.Helper()
 	if reflect.TypeOf(actual) != reflect.TypeOf(expected) {
-		t.Errorf("unexpected RSA private key, got %T, want %T", actual, expected)
+		t.Fatalf("unexpected RSA private key, got %T, want %T", actual, expected)
 	}
 	switch epk := expected.(type) {
 	case *rsa.PrivateKey:
@@ -157,7 +157,7 @@ func privateKeysEqual(t *testing.T, actual, expected crypto.Signer) {
 	case *ecdsa.PrivateKey:
 		require.True(t, epk.Equal(actual), "private keys should match")
 	default:
-		t.Errorf("unexpected RSA private key, got %T, want %T", actual, expected)
+		t.Fatalf("unexpected RSA private key, got %T, want %T", actual, expected)
 	}
 }
 

--- a/pkg/controller/common/certificates/ca_reconcile_test.go
+++ b/pkg/controller/common/certificates/ca_reconcile_test.go
@@ -153,9 +153,9 @@ func privateKeysEqual(t *testing.T, actual, expected crypto.Signer) {
 	}
 	switch epk := expected.(type) {
 	case *rsa.PrivateKey:
-		require.True(t, epk.Equal(expected))
+		require.True(t, epk.Equal(actual), "private keys should match")
 	case *ecdsa.PrivateKey:
-		require.True(t, epk.Equal(expected))
+		require.True(t, epk.Equal(actual), "private keys should match")
 	default:
 		t.Errorf("unexpected RSA private key, got %T, want %T", actual, expected)
 	}
@@ -413,9 +413,9 @@ func Test_buildCAFromSecret(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			ca := BuildCAFromSecret(context.Background(), tt.internalSecret)
 			if tt.wantCa == nil {
-				assert.Nil(t, ca)
+				require.Nil(t, ca)
 			} else {
-				assert.NotNil(t, ca)
+				require.NotNil(t, ca)
 				assert.True(t, ca.Cert.Equal(tt.wantCa.Cert), "certificates should be equal")
 				privateKeysEqual(t, ca.PrivateKey, tt.wantCa.PrivateKey)
 			}

--- a/pkg/controller/common/certificates/ca_reconcile_test.go
+++ b/pkg/controller/common/certificates/ca_reconcile_test.go
@@ -412,8 +412,12 @@ func Test_buildCAFromSecret(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			ca := BuildCAFromSecret(context.Background(), tt.internalSecret)
-			if !reflect.DeepEqual(ca, tt.wantCa) {
-				t.Errorf("CaFromSecrets() got = %v, want %v", ca, tt.wantCa)
+			if tt.wantCa == nil {
+				assert.Nil(t, ca)
+			} else {
+				assert.NotNil(t, ca)
+				assert.True(t, ca.Cert.Equal(tt.wantCa.Cert), "certificates should be equal")
+				privateKeysEqual(t, ca.PrivateKey, tt.wantCa.PrivateKey)
 			}
 		})
 	}

